pkg/agent/provisioning.go

```diff
@@ -1,24 +1,77 @@
 package agent
 
 import (
+	"fmt"
+
 	"github.com/Axway/agent-sdk/pkg/agent/handler"
 	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
 	"github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
+	mv1a "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
 	"github.com/Axway/agent-sdk/pkg/apic/provisioning"
+	"github.com/Axway/agent-sdk/pkg/migrate"
+	"github.com/Axway/agent-sdk/pkg/util/log"
 )
 
 // credential request definitions
 // createOrUpdateDefinition -
-func createOrUpdateDefinition(data v1.Interface) (*v1.ResourceInstance, error) {
+func createOrUpdateDefinition(data v1.Interface, marketplaceMigration migrate.Migrator) (*v1.ResourceInstance, error) {
 	if agent.agentFeaturesCfg == nil || !agent.agentFeaturesCfg.MarketplaceProvisioningEnabled() {
 		return nil, nil
 	}
-	return agent.apicClient.CreateOrUpdateResource(data)
+
+	var runMarketplaceMigration = false
```

Comment:
Can this migration check be moved to its own func?

Wy not checks for access request defs? What if an agent makes a blanket one like the credential request defs are?  such as api-key?

Reply:
It can be...
...and yes it should now check for ARDs